backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -169,8 +172,7 @@ public void removeStaff(int id) {
   public Optional<Staff> findStaffByEmail(String email) {
     return staffRepository.findAll().stream()
         .filter(staff -> staff
-            .getEmail()
-            .equalsIgnoreCase(email))
+            .getEmail().isPresent() && staff.getEmail().get().equalsIgnoreCase(email))
```

Comment:
this can also be written as
`staff.getEmail().filter(email::equalsIgnoreCase).isPresent()`
which saves one `getEmail` call. Not a big deal overall.
Filter on an optional will return the original optional if the predicate is true or an empty optional otherwise.

Reply:
No I like your way! I was trying to work out how to do this but ended up doing it this other way as a fallback. I like the functional filter chaining!